src/rtutils.c

```diff
@@ -999,7 +999,7 @@ static size_t jl_static_show_x_(JL_STREAM *out, jl_value_t *v, jl_datatype_t *vt
         n += jl_static_show_x(out, *(jl_value_t**)v, depth);
         n += jl_printf(out, ")");
     }
-    else if (jl_function_type && jl_isa(v, (jl_value_t*)jl_function_type)) {
+    else if (jl_function_type && jl_subtype(vt, (jl_value_t*)jl_function_type)) {
```

Comment:
I wonder why this doesn't fail the analyzegc pass, as this function is prohibited from here

Reply:
😱 
Which function is prohibited? The old code, or the new code, or both?

Reply:
I'm not sure why `jl_is_operator` isn't failing the analyzegc pass either, as that's also prohibited by our annotations here

Reply:
It turns out that we made a mistake in the first PR, and hadn't noticed this comment:
https://github.com/JuliaLang/julia/blob/8f54d34c02206f6d8976e6e7916b51055a0f7a19/src/rtutils.c#L644-L648

This PR is attempting to now follow the rules of that comment.

Reply:
The analyzer sometimes bails on big functions. This one definitely qualifies

Reply:
But yeah, you're technically not allowed to use jl_subtype (or jl_isa). You may look at the `->super` field of the datatype though and see if you hit function eventually.

Reply:
Thanks @Keno. I believe I've fixed this in 2e5f22c1d7, by implementing a while loop using `->super`, as you suggest.

Could you take another look when you have a moment? Thanks! :)